draft-irtf-cfrg-vdaf.md

```diff
@@ -3663,37 +3663,60 @@ way. The parameters for this circuit are summarized below.
 
 ### Prio3Sum
 
-| Parameter         | Value                                      |
-|:------------------|:-------------------------------------------|
-| `Valid`           | `Sum(Field128, bits)` (this section)       |
-| `Field`           | `Field128` ({{fields}})                    |
-| `PROOFS`          | `1`                                        |
-| `Xof`             | `XofTurboShake128` ({{xof-turboshake128}}) |
+| Parameter         | Value                                          |
+|:------------------|:-----------------------------------------------|
+| `Valid`           | `Sum(Field64, max_measurement)` (this section) |
+| `Field`           | `Field64` ({{fields}})                         |
+| `PROOFS`          | `1`                                            |
+| `Xof`             | `XofTurboShake128` ({{xof-turboshake128}})     |
 {: title="Parameters for Prio3Sum."}
 
 The next instance of Prio3 supports summing of integers in a pre-determined
-range. Each measurement is an integer in range `[0, 2^bits)`, where `bits` is an
-associated parameter.
+range. Each measurement is an integer in `range(max_measurement+1)`, where
+`max_measurement` is an associated parameter equal to the largest valid
+measurement.
+
+The range check is accomplished by encoding the measurement as a bit vector,
+encoding the measurement plus an offset as a bit vector, then checking that the
+two encoded integers are consistent. Let
+
+* `bits = max_measurement.bit_length()`, the number of bits needed to encode
+  the largest valid measurement
+
+* `offset = 2**bits - 1 - max_measurement`
+
+The first bit-encoded integer is the measurement itself. Note that only
+measurements between `0` and `2**bits - 1` can be encoded this way with
+`bits` bits. The second bit-encoded integer is the sum of the measurement
+and `offset`. Observe that only measurements between `-offset` and
+`max_measurement` inclusive can be encoded this way with `bits` bits.
```

Comment:
@divergentdave if true, this is not the intent of the construction, and it contradicts the statement above: 

>  Each measurement is an integer in `range(max_measurement+1)`, where associated parameter. `max_measurement` is an associated parameter equal to the largest valid measurement.

Reply:
This is referring specifically to the second encoded integer, but not the first encoded integer. We only get `range(max_measurement+1)` when combining the two, and checking they're consistent. Maybe this would be clearer?

> Observe that this sum can only be encoded this way if it is between `0` and `2**bits - 1`, which implies that the measurement is between `-offset` and `max_measurement`.

Reply:
Ah ok, that makes more sense! Yes that's clearer.